app/src/main/java/com/zegoggles/smssync/service/SmsBackupService.java

```diff
@@ -262,10 +265,12 @@ private void notifyAboutBackup(BackupState state) {
     }
 
     private void scheduleNextBackup() {
-        final long nextSync = getAlarms().scheduleRegularBackup();
-        if (nextSync >= 0) {
+        final Job nextSync = getAlarms().scheduleRegularBackup();
```

Comment:
shouldn't this be just a recurring job? in which case the logic doesn't have to be in the service any longer.

Reply:
I guess it could be - I just kept the original logic.
Although now it only runs when needed.
I've been running with this patch for a while, and have seen the job run even though no network service was available, so making it recurring would help with that issue.

Reply:
I hope this patch will help with reliability. Another thing I'm wondering is if the content provider observation could be used to replace the SMS broadcast listener. Have you looked at this?

Reply:
I have had the problems where backup was not performed and using this patch has removed the problem.
I haven't looked into using the content provider.

Reply:
Actually, if you don't schedule a new job, the JobService keeps the wakelock active. You're supposed to call onFinish otherwise.

https://github.com/jberkel/sms-backup-plus/pull/822/commits/dc7ea3bb82d83b23aeb56e05a5d10f5ac1249eed#diff-87977571b8d91982f3265c1edc629684R33